text/trace/0192-messaging-semantic-conventions-spec.md

```diff
@@ -0,0 +1,660 @@
+  +------------+                           .
+                                   +---------------------------+
+                               . . | Deliver                   |
+  +------------+               .   +-+------------+------------+
+  | Publish m2 | . . . . . . . .     | Process m1 |
+  +------------+                     +------------+------------+
+                                                  | Process m2 |
+                                                  +------------+
+```
+
+#### Single message producers, batch pull-based consumer with process spans
+
+```
+  PRODUCER                    CONSUMER
+
+  +------------+
+  | Publish m1 |. . . .       +-------------------------------------+
+  +------------+      .       | Ambient                             |
+                      .       +-+---------+-------------------------+
+                      . . . . . | Receive |
+  +------------+          .     +---------+------------+
+  | Publish m2 |. . . . . .               | Process m1 |
+  +------------+                          +------------+------------+
+                                                       | Process m2 |
+                                                       +------------+
+```
+
+#### Single message producers, batch pull-based consumer with manual settlement
+
+```
+  PRODUCER                    CONSUMER
+
+  +------------+
+  | Publish m1 |. . . .       +----------------------------------------+
+  +------------+      .       | Ambient                                |
+                      .       +-+---------+----------------------------+
+                      . . . . . | Receive |
+  +------------+          .     +---------+    +-----------+
+  | Publish m2 |. . . . . .                    | Settle m1 |
+  +------------+                               +-----------+-----------+
+                                                           | Settle m2 |
+                                                           +-----------+
+```
+
+## Future possibilities
+
+### Transport context propagation
+
+One possibility to seamlessly integrate producer/consumer and intermediary
+instrumentation in a flexible and extensible way would be the introduction of a
+second transport context layer in addition to the creation context layer.
+
+1. The _creation context layer_ allows correlating the producer with the
+   consumers of a message, regardless of intermediary instrumentation. The
+   creation context is created by the producer and must be propagated to the
+   consumers. It must not be altered by intermediaries.
+
+   This layer helps to model dependencies between producers and consumers,
+   regardless of the underlying messaging transport mechanism and its
+   instrumentation.
+2. An additional _transport context layer_ allows correlating the producer and
+   the consumer with an intermediary. It also allows to correlate multiple
+   intermediaries among each other. The transport context can be changed by
```

Comment:
The correlation should be message based. "Correlating intermediaries" sounds like broker framework related work. Do we want to provide tools to make that easier? Which ones?